MergerLogic/DataTypes/Gpkg.cs

```diff
@@ -73,26 +74,48 @@ public override void Reset()
             this._offset = 0;
         }
 
-        public override List<Tile> GetNextBatch(out string batchIdentifier)
+        public override List<Tile> GetNextBatch(out string currentBatchIdentifier, out string? nextBatchIdentifier, string? inCompletedBatchIdentifier, long? totalTilesCount)
         {
-            batchIdentifier = this._offset.ToString();
-            //TODO: optimize after IOC refactoring
-            int counter = 0;
-            List<Tile> tiles = this.Utils.GetBatch(this.BatchSize, this._offset)
-                .Select(t =>
+            lock (_locker)
+            {
+                if (inCompletedBatchIdentifier is not null)
```

Comment:
This is unnessecary and incorrect, offset is already set before `DoWork` here: [newData](https://github.com/MapColonies/GpkgMerger/blob/aadca3758fa37531cd734802567d4da538f67117/MergerCli/Process.cs#L128)
Remove `inCompletedBatchIdentifier` from function parameters.